src/structures/MessageComponentInteraction.js

```diff
@@ -49,6 +49,12 @@ class MessageComponentInteraction extends Interaction {
      * @type {InteractionWebhook}
      */
     this.webhook = new InteractionWebhook(this.client, this.applicationID, this.token);
+
+    /**
+     * The values selected, if the component which was interacted with was a select menu
+     * @type {string[]}
+     */
+    this.values = this.componentType === 'SELECT_MENU' ? data.data.values : null;
```

Comment:
If the property only exists when it's a SELECT_MENU, why not make a `SelectMenuMessageComponentInteaction` class?

The line 33 would need to be refactored so we don't resolve the component type twice, but I find it more ideal to define the properties that exist only in different component types, similar to why we have TextChannel, VoiceChannel, DMChannel, etc.

Reply:
I don't entirely disagree, but I'm definitely dropping the generic "MessageComponent" part of the naming convention if we do make these... well, not generic. `ButtonInteraction` and `SelectMenuInteraction` would fine imo.

Following from that change, what collectors do we build into Message and TextChannel? Do those also get separated?